metrix/src/app.rs

```diff
@@ -41,6 +41,89 @@ fn query_metric_route(
 ) -> Result<Json<Vec<Metric>>, BadRequest<String>> {
     let db_conn = establish_connection();
 
+    let filter_clause: String;
+    let result = build_filter_clause(offset, start_datetime, end_datetime, q);
+    match result {
+        Ok(o) => {
+            filter_clause = o;
+        },
+        Err(e) => {
+            return Err(e);
+        },
+    }
+
+    println!("### QUERY: SELECT * FROM metrics {};", filter_clause);
+
+    let query_string = format!("SELECT * FROM metrics {} ORDER BY id LIMIT 10", filter_clause);
+    let results = sql_query(query_string)
+        .load(&db_conn)
+        .expect("Error loading metrics");
+
+    Ok(Json(results))
+}
+
+#[get("/<aggregation>?<offset>&<start_datetime>&<end_datetime>&<q>&<bucket_count>")]
+fn aggregate_metrics_route(
+    aggregation: Option<&RawStr>,
+    offset: Option<&RawStr>,
+    start_datetime: Option<&RawStr>,
+    end_datetime: Option<&RawStr>,
+    q: Option<&RawStr>,
+    bucket_count: i32,
+) -> Result<Json<Buckets>, BadRequest<String>> {
+
+    let db_conn = establish_connection();
+
+    let filter_clause: String;
+    let result = build_filter_clause(offset, start_datetime, end_datetime, q);
+    match result {
+        Ok(o) => {
+            filter_clause = o;
+        },
+        Err(e) => {
+            return Err(e);
+        },
+    }
+
+    let start_timestamp= NaiveDateTime::parse_from_str(
+        &start_datetime.unwrap(),
+        &"%Y-%m-%dT%H:%M:%S".to_string()
+    ).ok().unwrap().timestamp();
+
+    let end_timestamp = NaiveDateTime::parse_from_str(
+        &end_datetime.unwrap(),
+        &"%Y-%m-%dT%H:%M:%S".to_string()
+    ).ok().unwrap().timestamp();
+
+    let bucket_size = (end_timestamp - start_timestamp) as f32 / bucket_count as f32;
```

Comment:
So weird we gotta account for floats here. If we set a granularity threshold it could be simpler...but probably not worth that optimization right now.

Reply:
Kinda does that naturally, `f32` is only so big. After a certainly point you could end up having things shift buckets if they're reaaaally close

Reply:
But yeah we could set a specific one by rounding if we want